### ISSUE TITLE
[Delegate] entities can be connected to a polymorphic model even without update permissions

### ISSUE THREAD

**Issue by ymc9** [zenstackhq/zenstack#1674]
## ZModel

```prisma
model User {
    id       String @id @default(cuid())
    email    String @unique @email @length(6, 32)
    password String @password @omit
    posts    Post[]

    // everybody can signup
    @@allow('create', true)

    // full access by self
    @@allow('all', auth() == this)
}

model Blog {
    id        String   @id @default(cuid())
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt

    post      Post?    @relation(fields: [postId], references: [id], onDelete: Cascade)
    postId    String?
}

model Post {
    id        String   @id @default(cuid())
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
    title     String   @length(1, 256)
    content   String
    published Boolean  @default(false)
    author    User     @relation(fields: [authorId], references: [id])
    authorId  String

    blogs     Blog[]

    type      String

    // allow read for all signin users
    @@allow('read', auth() != null && published)

    // full access by author
    @@allow('all', author == auth())

    @@delegate(type)
}

model PostA extends Post {
}

model PostB extends Post {
}
```

## TS

```typescript
const user = await prisma.user.create({
    data: { email: "abc@def.com", password: "password" },
  });

  const blog = await prisma.blog.create({
    data: {},
  });

  const enhancedDb = enhance(prisma, { user }, { logPrismaQuery: true });

  // the following call should be rejected because `Blog` model is not updateable
  await enhancedDb.postA.create({
    data: {
      content: "content",
      title: "title",
      blogs: {
        connect: {
          id: blog.id,
        },
      },
      author: {
        connect: {
          id: user.id,
        },
      },
    },
  });
```

**Comment by ymc9** [comment]
Reported by @eqqe

**Comment by ymc9** [comment]
Fixed in 2.5.0
